compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -47,7 +47,26 @@ use rustc_span::hygiene::HygieneDecodeContext;
 
 mod cstore_impl;
 
-crate struct MetadataBlob(MetadataRef);
+/// A reference to the raw binary version of crate metadata.
+/// A `MetadataBlob` internally is just a reference counted pointer to
+/// the actual data, so cloning it is cheap.
+#[derive(Clone)]
+crate struct MetadataBlob(Lrc<MetadataRef>);
```

Comment:
Would turning `MetadataRef` from a type alias of`OwningRef<Box<dyn Erased>, [u8]>` into a type alias of `OwningRef<Lrc<dyn Erased>, [u8]>` work too instead of this extra wrapper?

Reply:
I think I tried that and it required a more changes to the internal copy of the owning_ref crate. Maybe it would be a good idea to try and turn this into an `Arc` which then might allow us to switch to the crates.io version of owning_ref? But I'd rather not do that as part of this PR.

Reply:
An `Lrc` is an alias to either an `Rc` or `Arc` depending on if parallel rustc is enabled or not.

Reply:
I know. I vaguely remember that that made things more complicated than seemed necessary -- but that was a while ago and I didn't try very hard. Feel free to take a stab at implementing the `OwningRef<Lrc<dyn Erased>, [u8]>` approach and I'll rebase this PR on top of it.